src/engines/TimeToLive.js

```diff
@@ -0,0 +1,149 @@
+import DefaultHashTable from '../dataStructure/HashTable';
+import DoublyLinkedList from '../dataStructure/DoublyLinkedList';
+import * as hashTableProp from '../hashTableSymbol';
+
+function TimeToLive({ HashTable = DefaultHashTable, defaultTTL } = {}) {
+  const store = new HashTable();
+  const timePartition = new HashTable();
+  const timeIndexInterval = 5 * 60 * 1000; // milliseconds.
+
+  let lowestTimePartition = Date.now();
+
+  this.add = (key, value, ttl = defaultTTL) => {
```

Comment:
Just something we might want to consider. When we remove this ttl and only have the default ttl we have more performance optimization potential.